compiler/semstmts.nim

```diff
@@ -729,6 +729,16 @@ proc semRaise(c: PContext, n: PNode): PNode =
     var typ = n.sons[0].typ
     if typ.kind != tyRef or typ.lastSon.kind != tyObject:
       localError(n.info, errExprCannotBeRaised)
+    
+    # check if the given object inherits from Exception
+    var base = typ.lastSon
+    while true:
+      if base.sym.name.s == "Exception":
+        break
+      if base.lastSon == nil:
+        localError(n.info, errExprIsNoException)
```

Comment:
Good error message: Write out which object type does not inherit from exception.